src/app/core/entity/entity.spec.ts

```diff
@@ -99,4 +99,27 @@ describe("Entity", () => {
     expect(data.searchIndices).toContain("John");
     expect(data.searchIndices).toContain("Doe");
   });
+
+  it("can perform a shallow copy of itself", () => {
+    const id = "t1";
+    const entity: Entity = new Entity(id);
+    entity["value"] = 1;
+    const otherEntity = entity.copy();
+    expect(otherEntity).toEqual(entity);
```

Comment:
I think a assertion should also ensure, that the entity is copied rather than just assigned. A pointer (`otherEntity = entity`) would pass this test as well.